File: UVA100 - The 3n + 1 problem/main.py
# input bug!
import sys

def alg(n):
	count = 1;
	while n!=1:
		if n%2:
			n = n * 3 + 1
		else:
			n = n / 2
		count += 1
	return count

for line in sys.stdin:
	word_count = 0
	arr = line.split()
	while word_count < len(arr):
		i = int(arr[word_count])
		j = int(arr[word_count+1])
		word_count += 2
		print(i, j, end = "")
		if i > j:
			tmp = i
			i = j
			j = tmp
		ans = 0
		while True:
			tmp = alg(i)
			ans = max(tmp, ans)
			i += 1
			if i>j:
				break
		print("", ans)

File: UVA100 - The 3n + 1 problem/main.c
#include <stdio.h>

int alg(int n){
	int count = 1;
	while(n!=1){
		if(n%2) n = 3 * n + 1;
		else n = n / 2;
		count++;
	}
	return count;
}

int main(){
	int i, j;
	while(scanf("%d %d", &i, &j)!=EOF){
		printf("%d %d", i, j);
		if(i > j){
			int tmp = i;
			i = j;
			j = tmp;
		}
		int ans = 0;
		do{
			int tmp = alg(i);
			ans = (tmp > ans)?tmp:ans;
		}while(++i<=j);
		printf(" %d\n", ans);
	}
}